### PR TITLE
UI/DataTable: initialize with Order and Range

### PR DESCRIPTION
**Pull request by nhaagen**
https://mantis.ilias.de/view.php?id=41328

Optionally initialize a DataTable with Range and/or Order.

This includes the fix from https://github.com/ILIAS-eLearning/ILIAS/pull/8083
Also, I removed withNumberOfRows/getNumberOfRows from the interface in favour of (allready existing) get-/withRange.

### CODE REVIEW

**Review by klees — APPROVED** [review]
LGTM 2.

**Review by thibsy — COMMENTED** [review]
Hi @nhaagen,

Thx a lot for your contribution to the UI framework.

About the concrete changes, please answer the following questions:

- [x] `Data::withNumberOfRows()` usages: could you check if there are any usages of this method inside ILIAS? And, it it is not too much work, maybe directly replace these calls, so we can request reviews in this PR.

Otherwise this LGTM and should solve both mantis issues. Since we are removing an interface method, we need to go to JF before we can merge this.

Kind regards,
@thibsy

**Comment by nhaagen** [comment]
There were, in fact, but two occurences of "withNumberOfRows"; I replaced those.
@chfsx, maybe you want ot check on those.

File: components/ILIAS/ResourceStorage/classes/Collections/View/RequestToDataTable.php
Comment (chfsx) [inline components/ILIAS/ResourceStorage/classes/Collections/View/RequestToDataTable.php:104]: @nhaagen Wouldn't that mean that only the first X entries are displayed? Or would the pagination itself then update that? I'm just a little confused about the static “0”.

Comment (chfsx) [reply components/ILIAS/ResourceStorage/classes/Collections/View/RequestToDataTable.php:104]: @nhaagen Ok, I think I understand it now, it's really just about the default values :-) That's fine with me, thank you very much!

**Review by chfsx — APPROVED** [review]

**Review by thibsy — APPROVED** [review]
Thx @nhaagen for investigating and @chfsx for the quick review.

This LGTM now. I will merge this when accepted by JF.

Kind regards,
@thibsy

**Comment by thibsy** [comment]
@nhaagen just noticed there's something wrong with the code-style of the changed files. Could you amend this, before we merge? Thx!

**Comment by matthiaskunkel** [comment]
Jour Fixe, 30 SEP 2024: We accept the suggested changes for ILIAS 10/trunk.

**Pull request merged**
